dev/ci/.gitlab-ci.yml

```
@@ -1,69 +1,64 @@
+# ==========================================================================
+# Main GitLab CI Configuration for the global-workflow Project
+# ==========================================================================
+#
+# This is the primary CI/CD configuration file that orchestrates the GitLab CI
+# testing framework for the global-workflow project. It defines pipeline stages
+# and includes specialized configuration files for different testing aspects.
+#
+# The pipeline architecture supports:
+#   1. Multiple host environments (hera, gaeac6, etc.)
+#   2. Two primary testing modes:
+#      - PR validation via standard test cases
+#      - CTest-based tests triggered via GitHub API
+#
+# Included files:
+#   - .gitlab-ci-ctests.yml: Handles CMake/CTest-based testing framework
+#   - .gitlab-ci-cases.yml: Defines templates for standard experiment cases
+#   - .gitlab-ci-hosts.yml: Contains host-specific configurations and test matrices
+#                          that can be extended to support additional hosts
+#
+# The host configuration in .gitlab-ci-hosts.yml is designed to be easily 
+# extended with new computing platforms and allows per-host specification 
+# of which test cases to run.
+
 stages:
   - build
   - create_experiments
   - run_tests
+  - finalize
 
-# Global variables
 variables:
-  BUILD: 'TODAY'
-  GIT_CLONE_PATH: '${CI_BUILDS_DIR}/${BUILD}/global-workflow'
+  BUILD: ''
+  CI_RUN_PATH: ${CI_BUILDS_DIR}/${BUILD}/${CI_COMMIT_SHORT_SHA}
+  GIT_CLONE_PATH: '${CI_RUN_PATH}/global-workflow'
   HOMEGFS: ${GIT_CLONE_PATH}
-  RUNTESTS_DIR: ${CI_BUILDS_DIR}/${BUILD}/RUNTESTS
+  RUNTESTS_DIR: ${CI_RUN_PATH}/RUNTESTS
   GIT_DEPTH: 1
   RUNNER_SCRIPT_TIMEOUT: 6h
   RUNNER_AFTER_SCRIPT_TIMEOUT: 6h
+  # Controls pipeline behavior: CTests (true) or PR cases (false)
+  GITHUB_API_TRIGGER: ${GITHUB_API_TRIGGER:-"false"}
+  PR_NUMBER: ${PR_NUMBER:-"0"}
+
+# Include specialized pipeline configuration files
+include:
+  - local: 'dev/ci/.gitlab-ci-ctests.yml'  # CTest framework configuration
+  - local: 'dev/ci/.gitlab-ci-cases.yml'   # Standard test case templates
+  - local: 'dev/ci/.gitlab-ci-hosts.yml'   # Host-specific configurations
 
+.base_config:
+  variables:
+    GIT_STRATEGY: none
 
-# Build stage for the global workflow on compute nodes
-build:
+# Common build template for all modalities
+.build_template:
   variables:
     GIT_STRATEGY: clone
     GIT_SUBMODULE_STRATEGY: recursive
   stage: build
   script:
-    - echo "Using build directory ${HOMEGFS} (dated ${BUILD_DATE})"
-    - dev/ci/scripts/utils/ci_utils_wrapper.sh build_compute
+    - echo "Using build directory ${HOMEGFS}"
+    - dev/ci/scripts/utils/ci_utils.sh build_compute
     - sorc/link_workflow.sh
     - mkdir -p ${RUNTESTS_DIR}
```

Here and elsewhere a multiline `script` is used, consider using the literal `|` for readability:
```suggestion
    - |
      echo "Using build directory ${HOMEGFS}"
      dev/ci/scripts/utils/ci_utils.sh build_compute
      sorc/link_workflow.sh
      mkdir -p ${RUNTESTS_DIR}
```